eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java

```diff
@@ -140,7 +153,57 @@ public void failsOnDuplicateNonIdenticalVariant() throws Exception {
+                                                                       VARIANT_TYPE, VALIDATED);
+        ClusteredVariant duplicateFirstClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
+                                                                               VARIANT_TYPE, true);
+        ClusteredVariant fourthClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig_4", START_1,
+                                                                      VARIANT_TYPE, VALIDATED);
+        ClusteredVariant fifthClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig_5", START_1,
+                                                                      VARIANT_TYPE, VALIDATED);
+        DbsnpClusteredVariantEntity firstVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(firstClusteredVariant), firstClusteredVariant);
+        DbsnpClusteredVariantEntity secondVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION_2, hashingFunction.apply(secondClusteredVariant), secondClusteredVariant);
+        DbsnpClusteredVariantEntity thirdVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION_2 + 1, hashingFunction.apply(thirdClusteredVariant), thirdClusteredVariant);
+        DbsnpClusteredVariantEntity fourthVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION_2 + 2, hashingFunction.apply(fourthClusteredVariant), fourthClusteredVariant);
+        DbsnpClusteredVariantEntity duplicateVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(duplicateFirstClusteredVariant), duplicateFirstClusteredVariant);
+        DbsnpClusteredVariantEntity fifthVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION_2 + 3, hashingFunction.apply(fifthClusteredVariant), fifthClusteredVariant);
+        List<DbsnpClusteredVariantEntity> batch = Arrays.asList(firstVariant, secondVariant, thirdVariant,
+                                                                duplicateVariant, fourthVariant, fifthVariant);
+
+        // we expect an exception caused by the duplicate variant when inserting the batch
+        boolean exceptionThrown = false;
+        thrown.expect(BulkOperationException.class);
+        try {
+            dbsnpClusteredVariantWriter.write(
+                    batch);
+        } finally {
+            // 5 variants should have been inserted, and the duplicate one is not any of them
+            List<DbsnpClusteredVariantEntity> accessions = mongoTemplate.find(new Query(),
+                                                                              DbsnpClusteredVariantEntity.class);
+            assertEquals(5, accessions.size());
+            assertFalse(accessions.contains(duplicateVariant));
```

Comment:
To be make that no other weird thing has happened and because the number of objects is so low, could you please check the other 5 are contained in the collection?

Reply:
Done in test class refactor.